embeddings/pipeline/lighting_hps_pipeline.py

```diff
@@ -0,0 +1,173 @@
+from abc import ABC
+from dataclasses import dataclass, field
+from tempfile import TemporaryDirectory
+from typing import Any, Dict, Generic, Optional, Tuple
+
+from embeddings.hyperparameter_search.configspace import ConfigSpace, SampledParameters
+from embeddings.hyperparameter_search.lighting_configspace import (
+    LightingTextClassificationConfigSpace,
+)
+from embeddings.hyperparameter_search.parameters import ParameterValues
+from embeddings.pipeline.hf_preprocessing_pipeline import (
+    HuggingFaceTextClassificationPreprocessingPipeline,
+)
+from embeddings.pipeline.hps_pipeline import (
+    AbstractHuggingFaceOptimizedPipeline,
+    OptunaPipeline,
+    _HuggingFaceOptimizedPipelineBase,
+)
+from embeddings.pipeline.lightning_classification import LightningClassificationPipeline
+from embeddings.pipeline.pipelines_metadata import LightningClassificationPipelineMetadata
+
+
+@dataclass
+class _OptimizedLightingPipelineBase(
+    _HuggingFaceOptimizedPipelineBase[ConfigSpace], ABC, Generic[ConfigSpace]
+):
+    input_column_name: str
+    target_column_name: str
+
+    dataset_dir: TemporaryDirectory[str] = field(init=False, default_factory=TemporaryDirectory)
+    tmp_model_output_dir: TemporaryDirectory[str] = field(
+        init=False, default_factory=TemporaryDirectory
+    )
```

Comment:
both fields are temporary; should we unify the names? e.g. remove tmp_ from tmp_model_output_dir to add to dataset_dir

Reply:
I can add `tmp` to dataset_dir as this fields are `temporal`.  I will also change it in the flair hps pipeline

Reply:
Done